corehq/apps/export/tests/test_det_schema_generation.py

```diff
@@ -90,6 +90,17 @@ def setUpClass(cls):
         super().setUpClass()
         cls.export_instance = FormExportInstance.wrap(cls.get_json('form_export_instance_with_repeat'))
 
+    def test_main_table_not_selected(self):
+        self.export_instance.tables[0].selected = False
+        with tempfile.NamedTemporaryFile(mode='wb', suffix='.xlsx') as tmp:
+            generate_from_form_export_instance(self.export_instance, tmp)
+            wb = load_workbook(filename=tmp.name)
+            self.assertEqual(1, len(wb.worksheets))
+            repeat_ws = wb.worksheets[0]
+            self._check_repeat_worksheet(repeat_ws)
+
+        self.export_instance.tables[0].selected = True
```

Comment:
Is this part of the test? I would expect the last line of a test to be an assertion, not an assignment. Is this more than just a simple assignment?

Reply:
it's restoring the state of `self.export_instance` to what it was at the beginning of the test to avoid affecting other tests that reference it. `addCleanup` might be more appropriate but I thought this was fine

Reply:
added a comment in https://github.com/dimagi/commcare-hq/pull/27040/commits/d37bf38b1f13bf97f72026ad82885ea13aeffc37

Reply:
Hmm, this does not seem fine to me. `addCleanup` seems better than the comment since it will run even if the test fails.

Reply:
I'll do this in a followup since I agree it's better/cleaner.

I am curious though - and might just be a difference in the way we use language - from a practical perspective can you articulate why you feel it isn't "fine"? My thinking is that our Travis runs fail fast so it shouldn't ever surface in normal CI. Therefore the only time I see it potentially causing an issue on would be if someone is doing active development on these tests and gets temporarily confused by an error.

Reply:
https://github.com/dimagi/commcare-hq/pull/27071